docs: initial skeleton (#6)
rebornplusplus/chisel: docs/index.md

# Chisel

> **Chisel** is a tool for carving and cutting Debian packages.

Reviewer: ```suggestion
**Chisel** is a tool for carving and cutting Ubuntu packages.
```

I know it's what is on the README, but we want to avoid the "does it work with non-Ubuntu archives?" question

Author: Updated in https://github.com/rebornplusplus/chisel/pull/6/commits/1c903e9084030821bcd69eb56c1d153ea693a4ec.

> 

Reviewer: ```suggestion

It is built on the idea of **package slices** - minimal, complimentary, and loosely coupled sets of files, based on the package’s metadata and content. Slices are basically subsets of the Ubuntu packages, with their own content and set of dependencies to other internal and external slices.

```

Author: Added in https://github.com/rebornplusplus/chisel/pull/6/commits/1c903e9084030821bcd69eb56c1d153ea693a4ec, thanks.

> It can derive a minimal Ubuntu-like Linux distribution using a release
> database that defines "slices" of Debian packages. Slices enable users to
> cherry-pick just the files they need from the Ubuntu archives, and combine them
> to create a new root file system which can be packaged.
> 
> If you need a way to create a minimal root file system based on Ubuntu
> packages, Chisel might be for you. It handles package slice dependency,
> fetching Ubuntu Pro packages, and allows you to create a fully operational but
> minimal root file system for your application.
> 
> Chisel is particularly useful for developers who are creating minimal
> containers such as {{Rocks}}, or {{Docker}} images.

Reviewer: Did you get this content from an existing place?

```suggestion
Chisel is able to extract a highly customized and specialized _Slice_ of the
Ubuntu distribution, which one could see as a block of stone from which
we can carve and extract the small and relevant parts we need to run
our applications.

It operates like a package manager, but for package slices, thus being
particularly useful for supporting developers in the creation of
smaller but equally functional container images.
```

Author: The first paragraph's from my discourse post. Others, just wrote it up.

Author: Updated in https://github.com/rebornplusplus/chisel/pull/6/commits/1c903e9084030821bcd69eb56c1d153ea693a4ec, thanks.

---------

## In this documentation

````{grid} 1 1 2 2

```{grid-item-card} [Tutorial](tutorial/getting-started)

> **Start here**: a hands-on introduction to Chisel for new users

Reviewer: It would be nicer if the beginnings and dividers were consistent. E.g. "start here**:**" vs "technical information **-**" vs "discussion and clarification "

Author: Updated, with inspiration from the Rocks docs.

```

```{grid-item-card} [How-to guides](how-to/index)

**Step-by-step guides** covering key operations and common tasks
```

````

````{grid} 1 1 2 2
:reverse:

```{grid-item-card} [Reference](reference/index)

**Technical information** - specifications, syntax, architecture
```

```{grid-item-card} [Explanations](explanation/index)

> **Discussion and clarification** of key topics

Reviewer: these subtitles need to be expanded a little bit more. Atm, they state the obvious. E.g. "start here: a hands-on intro...". 

These were probably inspired by the Pebble docs, but I think we should also consider the Rockcraft docs as an example.

Author: Updated, with inspiration from the Rockcraft docs.
